java/gui/src/main/java/com/squareup/subzero/actions/FinalizeWallet.java

```diff
@@ -76,10 +80,27 @@ public static CommandResponse.FinalizeWalletResponse finalizeWallet(
     // Save the encrypted pubkeys to the wallet
     Wallet newWallet = Wallet.newBuilder(wallet)
         .addAllEncryptedPubKeys(request.getFinalizeWallet().getEncryptedPubKeysList())
+        .setCurrency(getCurrencyFromFinalizeResponse(iresp))
         .build();
     walletLoader.save(request.getWalletId(), newWallet);
 
     // Build response
     return CommandResponse.FinalizeWalletResponse.newBuilder().setPubKey(iresp.getPubKey()).build();
   }
+
+  private static Wallet.Currency getCurrencyFromFinalizeResponse(InternalCommandResponse.FinalizeWalletResponse iresp) {
```

Comment:
nit: not introduced by you but `iresp` 🙄

Reply:
What's wrong with `iresp`?